### DISCUSSION TITLE
Qtile + VSCode bug? New terminal opens in wrong window

### DISCUSSION THREAD

**Discussion by pawamoy** [qtile/qtile#4713] [Q&A]
Really random guess here, but I've always had this issue in VSCode and only just now reported it. It happens often and is very reproducible, so if it's still not fixed it's probably because very few users experience it, and so I guess it might because of something Qtile does with windows id/titles.

For reference: https://github.com/microsoft/vscode/issues/206548

Anyone else experiences this issue?

**Comment by elParaguayo** [comment]
I can recreate this behaviour.

I assume you're using the `Max` layout. By default, all windows other than the visible window are hidden (unmapped). However, if you disable this behaviour by setting `only_focused=False`, the terminal opens in the correct window.

I'm not sure why vscode decides to open the terminal in a hidden window though...

**Reply by pawamoy** [reply]
Ah, super interesting, thanks! I'll set `only_focused=False` in the meantime, thanks a lot.

**Reply by pawamoy** [reply]
Turns out I quickly switched back to `only_focused=True` because I have issues otherwise (not sure which issues anymore, I'll try again).

**Reply by pawamoy** [reply]
OK just tried again, and for example if I have both Firefox and Signal Desktop on the same screen, with `only_focused=False`, alt-tabbing to Signal doesn't bring it to the front. The title in Qtile's bar is updated to Signal, but I still see Firefox. I can bring Signal to the front by clicking on its icon in the sys tray, but alt-tabbing again resumes to the previous situation.

It looks like the culprit is Firefox. Same issue with Firefox + Terminator, while Signal + Terminator works fine.

Anyway `only_focused=False` is currently unusable to me so I have to switch back to `only_focused=True`, which triggers the VSCode issue but is at least manageable :smile:

**Reply by ervinpopescu** [reply]
QQ: what does `alt-tabbing` mean? What do you use? https://github.com/sagb/alttab?

**Reply by pawamoy** [reply]
Oh, sorry, I just meant hitting keys Alt + Tab at the same time (common keybinding to switch between open windows). In Qtile I use `lazy.layout.next()` for alt+tab and `lazy.layout.previous()` for alt+shift+tab.

**Reply by ervinpopescu** [reply]
Well, see, that sounds like an important information to me, since it does look like when calling the layout commands the window is not properly focused (why only with Firefox though, I do not know). I'd let @tych0 or @elParaguayo chime in on this, though.

**Reply by pawamoy** [reply]
Thanks @ervinpopescu. I should probably open a new issue.
